src/binary/decimal.rs

```diff
@@ -48,7 +49,7 @@ where
 {
     fn encode_decimal(&mut self, decimal: &Decimal) -> IonResult<usize> {
         // 0d0 has no representation, as per the spec.
-        if decimal == &DECIMAL_POSITIVE_ZERO {
+        if decimal.ion_eq(&DECIMAL_POSITIVE_ZERO) {
             return Ok(0);
         }
```

Comment:
🗺️ `IonEq` is stricter than `PartialEq`; `==` will ignore differences in sign/exponent when comparing two `Decimal` zeros.